app/models/personal_detail.rb

```diff
@@ -21,19 +26,26 @@ class PersonalDetail
   validates :gender, presence: true, inclusion: { in: Trainee.genders.keys }
   validate :nationalities_cannot_be_empty
 
-  def initialize(trainee:)
+  def initialize(trainee)
     @trainee = trainee
-    super(trainee.attributes.slice(*FIELDS).merge(nationality_ids: trainee.nationality_ids))
+    super(fields)
+  end
+
+  def fields
+    trainee.attributes.slice(*FIELDS).merge(nationality_ids: trainee.nationality_ids)
   end
 
 private
 
+  attr_reader :attributes
+
   def nationalities_cannot_be_empty
     return unless trainee.nationalities.empty?
 
-    errors.add(
-      :nationality_ids,
-      :empty_nationalities,
-    )
+    errors.add(:nationality_ids, :empty_nationalities)
+  end
+
+  def update_trainee
+    trainee.save! if errors.empty?
   end
```

Comment:
```
thing = PersonalDetail.new
```

I'm not too sure that I agree with this side effects, it feels error prone in longer terms...
```
thing.valid?
```

So maybe 
```suggestion
  def save
     valid? && trainee.save!
  end
```

Reply:
We should follow the `ActiveRecord` conventions here. So it should either call `trainee.save` and return a boolean if it fails or be named `save!` and raise if it fails. Otherwise I think `save` is correct.

Reply:
I was thinking. Should we be putting thise "form" objects into a separate directory, say `app/forms`? I think it would help make the project more readable. I would even goes as far as saying form objects have a certain signature name such as  `PersonalDetailForm` or `Form::PersonalDetail`.

Reply:
Happy to make those changes @defong . @John Henderson Regarding treating them as form objects, happy to go down that road if we’re only ever going to handle these objects around the view layer. It will feel a bit weird using them beyond this (if we need to). Also, think it should be done in a different PR as there’d be a lot to change and deal with in current PR

Reply:
changes applied in https://github.com/DFE-Digital/register-trainee-teachers/pull/161/commits/0ed3bc1d2e73aecce233ca59c5d5e5683429a6f9